docs/guides/image-update.md

```diff
@@ -594,286 +594,191 @@ some common examples for the most popular cloud provider docker registries.
 
-```yaml
-spec:
-  secretRef:
-    name: gcr-credentials
-```
+Flux will apply the Deployment and it will use the workload identity for that Pod to regularly fetch GCR tokens into your configured `KUBE_SECRET` name.
+Reference the `KUBE_SECRET` value from any `ImageRepository` objects for that GCR registry.
+
+This example uses the `fluxcd/flux2` github archive as a remote base, but you may copy the [./manifests/integrations/registry-credentials-sync/gcp]
+folder into your own repository or use a git submodule to vendor it if preferred.
 
 #### Using a JSON key [long-lived]
 
 !!! warning "Less secure option"
-    From [Google documentation on authenticating container registry](https://cloud.google.com/container-registry/docs/advanced-authentication#json-key)
+    From [Google documentation on authenticating container registry][JSON key file Authentication method]
     > A user-managed key-pair that you can use as a credential for a service account.
     > Because the credential is long-lived, it is the least secure option of all the available authentication methods.
     > When possible, use an access token or another available authentication method to reduce the risk of
     > unauthorized access to your artifacts. If you must use a service account key,
     > ensure that you follow best practices for managing credentials.
 
-A Json key doesn't expire, so we don't need a cronjob,
-we just need to create the secret and reference it in the ImagePolicy.
+A JSON key doesn't expire, which makes the need for a deployment to refresh tokens redundant. All that is needed is to create the secret and reference it in the ImagePolicy.
 
-First, create a json key file by following this
-[documentation](https://cloud.google.com/container-registry/docs/advanced-authentication).
-Grant the service account the role of `Container Registry Service Agent`
-so that it can access GCR and download the json file.
+Follow steps 1-3 of the official GCP documentation regarding [JSON key file Authentication method]. This will guide you through creating a service account, and obtaining a JSON key file that can access GCR.
 
-Then create a secret, encrypt it using [Mozilla SOPS](mozilla-sops.md)
-or [Sealed Secrets](sealed-secrets.md) , commit and push the encypted file to git.
+Generate a Kubernetes secret manifest with kubectl:
 
 ```sh
-kubectl create secret docker-registry <secret-name> \
-  --docker-server=<GCR-REGISTRY> \ # e.g gcr.io
-  --docker-username=_json_key \
-  --docker-password="$(cat <downloaded-json-file>)"
+$ kubectl create secret docker-registry <secret-name> \
+    --docker-server=<GCR-REGISTRY> \ # e.g gcr.io
+    --docker-username=_json_key \
+    --docker-password="$(cat <downloaded-json-file>)" \
+    --dry-run=client \
+    -o yaml > registry-credentials.yaml
 ```
+Encrypt it using [Mozilla SOPS] or [Sealed Secrets]. Once encrypted commit and push the encrypted file to git.
```

Comment:
Will people have to set anything up in Flux for the encrypted secret to be usable?